### PR TITLE
Makefile for cross compilation

### PR DESCRIPTION
**Pull request by sriumcp**
Partially addresses #1098 

The updated Makefile is modeled after the excellent Makefile in the Helm project.

Signed-off-by: Srinivasan Parthasarathy <spartha@us.ibm.com>

### CODE REVIEW

**Comment by sriumcp** [comment]
This PR is completed and ready for review. However, it is intended to be merged after #1107

File: Makefile [hunk 1]
```diff
@@ -1,33 +1,128 @@
...
+# ------------------------------------------------------------------------------
+#  release
+
+.PHONY: build-cross
+build-cross: LDFLAGS += -extldflags "-static"
+build-cross: $(GOX)
+	GOFLAGS="-trimpath" GO111MODULE=on CGO_ENABLED=0 $(GOX) -parallel=3 -output="_dist/{{.OS}}-{{.Arch}}/$(BINNAME)" -osarch='$(TARGETS)' $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LDFLAGS)' ./
+
+.PHONY: dist
+dist:
```
Comment (kalantar) [inline Makefile]: should be dependent on build-cross; otherwise get error since `_dist` may not exist.

**Review by kalantar — CHANGES_REQUESTED** [review]
Minor change suggested

File: Makefile (hunk 1 above)
Comment (sriumcp) [reply Makefile]: Fixed.

File: Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@ BINDIR      := $(CURDIR)/bin
 INSTALL_PATH ?= /usr/local/bin
 DIST_DIRS   := find * -type d -exec
 TARGETS     := darwin/amd64 darwin/arm64 linux/amd64 linux/386 linux/arm linux/arm64 linux/ppc64le linux/s390x windows/amd64
-TARGET_OBJS ?= darwin-amd64.tar.gz darwin-amd64.tar.gz.sha256 darwin-amd64.tar.gz.sha256sum darwin-arm64.tar.gz darwin-arm64.tar.gz.sha256 darwin-arm64.tar.gz.sha256sum linux-amd64.tar.gz linux-amd64.tar.gz.sha256 linux-amd64.tar.gz.sha256sum linux-386.tar.gz linux-386.tar.gz.sha256 linux-386.tar.gz.sha256sum linux-arm.tar.gz linux-arm.tar.gz.sha256 linux-arm.tar.gz.sha256sum linux-arm64.tar.gz linux-arm64.tar.gz.sha256 linux-arm64.tar.gz.sha256sum linux-ppc64le.tar.gz linux-ppc64le.tar.gz.sha256 linux-ppc64le.tar.gz.sha256sum linux-s390x.tar.gz linux-s390x.tar.gz.sha256 linux-s390x.tar.gz.sha256sum windows-amd64.zip windows-amd64.zip.sha256 windows-amd64.zip.sha256sum
+
```
Comment (kalantar) [inline Makefile:5]: Probably a good  choice to reduce the set of targets!

**Review by kalantar — APPROVED** [review]
looks good

File: Makefile (hunk 2 above)
Comment (sriumcp) [reply Makefile:5]: target_objs weren't being used in the makefile so I took them out. Helm uses them for signing/checksum.

**Pull request merged**